### PR TITLE
TYP: core.internals

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

Annotating Block as `Union[np.ndarray, ExtensionArray]`, mypy gives tons of complaints about EA not having `.T`, so this adds that, then updates tests as appropriate

### CODE REVIEW

File: pandas/core/arrays/base.py [hunk 1]
```diff
@@ -1081,6 +1081,19 @@ def _formatter(self, boxed: bool = False) -> Callable[[Any], Optional[str]]:
     # Reshaping
     # ------------------------------------------------------------------------
 
+    def transpose(self, *axes):
```
Comment (jreback) [inline pandas/core/arrays/base.py:1084]: can you type these

File: pandas/core/arrays/base.py [hunk 2]
```diff
@@ -1081,6 +1081,19 @@ def _formatter(self, boxed: bool = False) -> Callable[[Any], Optional[str]]:
...
+        """
+        Return a transposed view on this array.
+
+        Because ExtensionArrays are always 1D, this is a no-op.  It is included
```
Comment (jreback) [inline pandas/core/arrays/base.py:1088]: does this method get listed in the api?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/arrays/base.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/core/arrays/base.py:1088]: will add to the class-level docstring

Comment (jreback) [reply pandas/core/arrays/base.py:1088]: ok

Comment (jbrockmendel) [reply pandas/core/arrays/base.py:1088]: added it to the docstring and got docbuild trouble.  im OK with skipping it since its a no-op.

**Review by jreback — APPROVED** [review]

File: pandas/core/arrays/base.py (hunk 2 above)
Comment (jreback) [reply pandas/core/arrays/base.py:1088]: pls add to list then, this is public API

**Pull request merged**